docs/userguide/CDN/USERGUIDE.md

```diff
@@ -0,0 +1,289 @@
+
+
+```php
+$cdnService = $client->cdnService();
+```
+
+## Services
+
+A service represents your web application that has its content cached to the edge nodes.
+
+### Create a service (to represent your web application)
+
+
+This operation takes one parameter, an associative array, with the following keys:
+
+| Name | Description | Data type | Required? | Default value | Example value |
+| ---- | ----------- | --------- | --------- | ------------- | ------------- |
+| `name` | A human-readable name for the service. This name must be unique. | String | No | `null` | `acme_site` |
+| `domains` | An array of associative arrays, each specifying a domain name for your service. | Array of associative arrays | Yes | `null` | `array( array( 'domain' => 'www.acme.com' ), array ( 'domain' => 'acme.com' ) )` |
+| `origins` | An array of associative arrays, each specifying an origin server for your service. The `port`, `ssl`, and `rules` elements for each origin server are optional. | Array of associative arrays | Yes | `null` | `array( array( 'origin' => 'origin.acme.com', 'port' => 80, 'ssl' => false, 'rules' => array() ) )` |
+| `flavorId` | The flavor used to configure this service. Use the [list flavors](#list-flavors) operation to retrieve a list of all available flavors. | String | Yes| `null` | `cdn` |
+| `restrictions` | An array of associative arrays, each specifying a restriction for who can or cannot access content from the CDN cache. | Array of associative arrays | No | `null` | `array( array( 'name' => 'website only', 'rules' => array( array( 'name' => 'mywebsite.com', 'httpHost' => 'www.mywebsite.com' ) ) ) )` |
+| `caching` | An array of associative arrays, each specifying a caching rule for your service's assets. | No | `null` | `array( array( 'name' => 'default', 'ttl' => 3600 ) )` |
+
+You can create a service as shown in the following example:
+
+```php
+$service = $cdnService->createService(array(
+    'name'     => 'acme_site',
+    'domains'  => array(
+        array(
+            'domain' => 'www.acme.com'
+        ),
+        array(
+            'domain' => 'acme.com'
+        )
+    ),
+    'origins'  => array(
+        array(
+            'origin' => 'origin.acme.com'
+        )
+    ),
+    'flavorId' => 'cdn'
+));```
+
+[ [Get the executable PHP script for this example](/samples/CDN/create-service.php) ]
+
+### List Services
+
+You can list all the services you have created as shown in the following example:
+
+```php
+$services = $cdnService->listServices();
+foreach ($services as $service) {
+    /** @var $service OpenCloud\CDN\Resource\Service **/
+}
+```
+
+[ [Get the executable PHP script for this example](/samples/CDN/list-services.php) ]
+
+### Get a service
+
+You can retrieve a specific service by using that service's ID, as shown in the following example:
+
+```php
+$service = $cdnService->getService('0e09ad12-2bfe-4607-80fd-116fa68d9c79');
```

Comment:
Hmm. That's not what we are doing in other docs. See https://github.com/rackspace/php-opencloud/blob/working/docs/userguide/Networking/USERGUIDE.md#get-a-network, https://github.com/rackspace/php-opencloud/blob/working/docs/userguide/Orchestration/USERGUIDE.md#get-stack, https://github.com/rackspace/php-opencloud/blob/working/docs/userguide/ObjectStore/USERGUIDE.md#get-container-details. If we do this here we should do it in other docs too.

Also, would you replace just IDs generated by the server but leave other sample data (provided by the client) in these docs?